### ISSUE TITLE
"Emit infantry on destruction" trait / make SpawnActorOnDeath emit multiple units

### ISSUE THREAD

**Issue by clemty** [OpenRA/OpenRA#9548]
TD used to spawn actors when buildings were destroyed (e.g. Commando blows up a SAM site: two minigunners appear); there apparently exists an `EmitInfantryOnSell` trait, but there is no such thing as `EmitInfantryOnDestruction`.

`SpawnActorOnDeath` is not listed on https://github.com/OpenRA/OpenRA/wiki/Traits , but it seems it can only emit a single actor:

```
SpawnActorOnDeath:
  Actor: e1
```

```
EmitInfantryOnSell:
  ActorTypes: e6,e1,e1,e1
```

Also the traits are named inconsistently (`ActorTypes` & `Actor`, ProductionAirdrop uses `ActorType`), and wouldn't `SpawnActorOnSell` be better? (I guess it's possible to emit vehicles as well)

**Comment by penev92** [comment]
"Yes" to `SpawnActorsOnSell` with an `Actors` property and "yes" to `SpawnActorsOnDeath` with an `Actors` property.
Also `SpawnActorOnDeath` is not on the traits page because it was just recently added and that's the traits page for the release.

**Comment by abcdefg30** [comment]
Isn't this doable now that you can have multiple instances of `SpawnActorOnDeath`?

**Comment by Phrohdoh** [comment]
@abcdefg30 Yes that is possible, but I don't see a reason not to fulfill this request.
`SpawnActorOnDeath` being conditional and allowing multiple Actor types to be provided is the most flexible without requiring many `SpawnActorOnDeath` instances.

This needs discussion and a decision on how we're going to supply spawn offsets.

The path of least breakage would be `Offset` being refactored into `CVec[] Offsets` with a 1:1 relationship with `Actors`.

**Comment by pchote** [comment]
The issue with a 1:1 relationship is that we don't have a way for specifying cell+subcell pairs.  It would be simpler IMO to define just the footprint of valid cells (e.g. `CVec[] Offsets`) and then spawn the units randomly within those.

In any case this should be sharing at least the user facing behaviour but ideally also the code with `SpawnActorsOnSell`. We could pull out a base class like we do for `AffectsShroud` or have a single trait that has an enum for `EmitType.Sell`, `EmitType.Destroyed`, etc.

**Comment by DeadSeriousBros** [comment]
I know this thread is old but I figured out how to spawn multiple actors when a unit or vehicle is destroyed.

**Instead of doing this:**
SpawnActorOnDeath:
 Actor: E1

**Do this:**
SpawnActorOnDeath@1:
 Actor: E1
SpawnActorOnDeath@2:
 Actor: E1
SpawnActorOnDeath@3:
 Actor: E1
SpawnActorOnDeath@4:
 Actor: E1

Keep adding up until you have reached the amount of units you want to spawn at one time!
I figured this out by looking at the default coding and noticing that they also add a @# for other Commands/Traits.
For more Property settings take a look at https://github.com/OpenRA/OpenRA/wiki/Traits#spawnactorondeath
Hope this helps anyone that may come across this thread like I did in search of spawning more than one actor on a unit(s) or vehicle(s) death! 😀👍
